app/src/main/res/values/strings.xml

```diff
@@ -68,6 +68,9 @@
     <string name="load_failure_explanation_dmca">This repository was taken down for legal reasons.\nPlease see %1$s for details.</string>
     <string name="load_failure_explanation_parsing">There was an error while parsing GitHub\'s response. You can use the button below to report this issue to the app developers.</string>
     <string name="load_failure_explanation_with_reason">Loading the data from GitHub failed. GitHub told us the following reason:\n%1$s</string>
+    <string name="snackbar_generic_error">%1$s. Please check your Internet connection.</string>
```

Comment:
I'm not sure about the 'please check your connection' text. It's probably helpful for the first time, but probably also redundant (to annoying) afterwards.
Also, why capitalize 'internet'?

Reply:
> Also, why capitalize 'internet'?

That's likely me being used to writing it using the Italian spelling :sweat_smile: 

Maybe something along the lines of _"This was likely caused by a connection error."_ would fit better?

Reply:
I think I prefer the original string (`Please check your internet connection.`) to that.
To avoid showing it unnecessarily, how about only showing that specific message in case of `IOException` being thrown, and omitting it in case of other exceptions?

Reply:
I think that's a good idea. I'm not a big of fan of the "X failed" message without any explanation though, even if it should rarely occur in practice. In those cases, maybe we could replace the "Copy error message" with a button to copy the stack trace and encourage the user to report the issue?

Reply:
That might be a good idea, or might cause similar spam as in #1332 ... not sure yet ;-)

Reply:
Agreed, I think that's OK.

Reply:
I have rebased and pushed the requested changes and added the ability to copy the error stack trace in case of application errors. I can confirm it works as expected because I've caught a JSON exception due to a GH error response not matching the `ClientErrorResponse` class in the bindings.
<img src="https://github.com/user-attachments/assets/878eb274-ac2a-4b9c-8569-fed94732e173" width="430" />

I have also changed the descriptive error snackbars to disappear after 8 seconds instead of being shown indefinitely.
Let me know what you think.